ninja-core/src/main/java/ninja/Route.java

```diff
@@ -125,10 +123,12 @@ public boolean matches(String httpMethod, String uri) {
         Matcher m = regex.matcher(uri);
 
         if (m.matches()) {
-            Iterator<String> it = this.parameters.keySet().iterator();
-            for (int i = 1; i < m.groupCount() + 1; i++) {
-                String parameterName = it.next();
-                map.put(parameterName, m.group(i));
+            if (this.parameters != null) {
```

Comment:
Another idea. What about returning an empty LinkedList when RouteParameter.parse(uri) does not return anything? Then we don't need any null checks... The test is still cool to make sure stuff works of course...

Not 100% sure but RouteParameter.parse(uri) is only called once when building the Route, so there should not be a performance impact of that...